File: packages/nitro-renderer/src/room/object/visualization/data/DirectionalOffsetData.ts
export class DirectionalOffsetData {
    private _offsetX: Map<number, number> = new Map();
    private _offsetY: Map<number, number> = new Map();

    public getXOffset(direction: number, defaultX: number): number {
        return this._offsetX.get(direction) ?? defaultX;
    }

    public getYOffset(direction: number, defaultY: number): number {
        return this._offsetY.get(direction) ?? defaultY;
    }

    public setDirection(direction: number, offsetX: number, offsetY: number): void {
        this._offsetX.set(direction, offsetX);
        this._offsetY.set(direction, offsetY);
    }
}

File: packages/nitro-swift/Sources/NitroRoom/Furniture/DirectionalOffsetData.swift
import Foundation

/// Swift port of `DirectionalOffsetData` (packages/nitro-renderer/src/room/object/visualization/data/DirectionalOffsetData.ts).
public final class DirectionalOffsetData {
    private var offsetX: [Int: Double] = [:]
    private var offsetY: [Int: Double] = [:]

    public func getXOffset(_ direction: Int, defaultX: Double) -> Double { offsetX[direction] ?? defaultX }
    public func getYOffset(_ direction: Int, defaultY: Double) -> Double { offsetY[direction] ?? defaultY }

    public func setDirection(_ direction: Int, offsetX: Double, offsetY: Double) {
        self.offsetX[direction] = offsetX
        self.offsetY[direction] = offsetY
    }
}
